crates/mun_runtime/src/test.rs

```diff
@@ -298,3 +298,55 @@ fn hotreloadable() {
     );
     assert_invoke_eq!(i64, 10, driver, "main");
 }
+
+#[test]
+fn compiler_vaild_ut8() {
+    use std::ffi::CStr;
+    use std::str;
+
+    let mut driver = TestDriver::new(
+        r#"
+    fn foo(n:int):bool { true }
+    fn main():int { if foo(1) { 5 } else { 4 } }
+    "#,
+    );
+
+    let foo_func = driver.runtime.get_function_info("foo").unwrap();
+    let main_func = driver.runtime.get_function_info("main").unwrap();
+
+    // Check foo name for valid utf8
+    assert_eq!(
+        str::from_utf8(&unsafe { CStr::from_ptr(foo_func.signature.name).to_bytes() }).is_err(),
+        false
+    );
+    // Check foo arg type for valid utf8
+    assert_eq!(
+        str::from_utf8(&unsafe { CStr::from_ptr((*foo_func.signature.arg_types).name).to_bytes() })
+            .is_err(),
+        false
+    );
+    // Check foo return type for valid utf8
+    assert_eq!(
+        str::from_utf8(&unsafe {
+            CStr::from_ptr((*foo_func.signature.return_type).name).to_bytes()
+        })
+        .is_err(),
+        false
+    );
+
+    // Check main name for valid utf8
```

Comment:
In essence, this and the following tests are the same as the tests for `foo_func`s tests, so they are unnecessary. If the logic is valid for one test case (that runs through all code paths) another test case will give the same results.

Since Mun doesn't require a main function, you can actually remove the `main` function from the example and remove the corresponding assertions.

Reply:
Good remark!